### ISSUE TITLE
[Bug]: Empty "assume_role.role_arn" results in provider error and fails to plan/apply

### ISSUE THREAD

**Issue by evan-cleary** [hashicorp/terraform-provider-aws#39296]
### Terraform Core Version

1.4.7

### AWS Provider Version

5.67.0

### Affected Resource(s)

* all

### Expected Behavior

Provider previously ignored assume role blocks when `role_arn` was left as empty string `""` or `null`.

We depend on this as we optionally configure assume_role in our terraform using configurations like:

```terraform
  assume_role {
    role_arn = var.assume_role
    ...
  }
```
and 

```terraform
assume_role {
   role_arn = var.assume_role
   duration = "15m"
}
```

### Actual Behavior

Provider now fails to plan/apply saying the role_arn is required.

### Relevant Error/Panic Output Snippet

```shell
Error: Missing required argument 
  with provider["registry.terraform.io/hashicorp/aws"], 
  on main.tf line 11, in provider "aws": 
  11: provider "aws" { 
The argument "role_arn" is required, but no definition was found.
```

### Terraform Configuration Files

```terraform
terraform {
  required_providers {
    aws = {
      version = "5.67.0"
      source = "hashicorp/aws"
    }
  }
}

provider "aws" {
  region = "us-east-1"
  assume_role {
    role_arn = ""
  }
}

data "aws_region" "current" {}
```

```terraform
terraform {
  required_providers {
    aws = {
      version = "5.67.0"
      source = "hashicorp/aws"
    }
  }
}

provider "aws" {
  region = "us-east-1"
  assume_role {
    role_arn = null
    duration = "15m"
  }
}

data "aws_region" "current" {}
```

### Steps to Reproduce

1. Run the above configuration(s)

### Important Factoids
The majority of our terraform configurations support conditionally using an assume_role to support our local development and the CI pipelines, this change has broken conditional assume roles for any configuration set up prior to the introduction of dynamics.

### References

This appears to be the result of: https://github.com/hashicorp/terraform-provider-aws/pull/39255

Comments within the schema change show intentional of backwards compatibility with empty `assume_role.role_arn` attributes
https://github.com/hashicorp/terraform-provider-aws/pull/39255/files#diff-cddd9e71748414f0ca51a47f3c35f8c45522d41c9477d3cfae3374dd656610f1R192

But then in implementation requirement of that property is enforced
https://github.com/hashicorp/terraform-provider-aws/pull/39255/files#diff-58d6a027753b50994deb7e11e4a99dde423f35844986019bd9cea5e0c94aba22R793-R798

### Would you like to implement a fix?

None

**Comment by quentin9696** [comment]
Hi, 

To have a cleaner TF syntax, this works with `5.67.0`

``` hcl
dynamic "assume_role" {
    for_each = var.aws_assume_role != null ? [1] : []

    content {
      role_arn = var.aws_assume_role
    }
  }

**Comment by gdavison** [comment]
This will be resolved by #39328. We're planning to release a bug fix update this Monday, 16 September.

In addition to the workarounds listed above, you can also set the value of `var.aws_assume_role` to `null` instead of `""`. For example:

```terraform
variable "var.aws_assume_role" {
  type    = string
  default = null
}
```

**Comment by iNoahNothing** [comment]
I am getting the same error even with with following config that defaults `role_arn` to `null`
```tf
provider "aws" {
  profile                  = var.aws_profile
  region                   = var.region
  shared_credentials_files = [pathexpand(var.shared_credentials_file)]

  assume_role {
    role_arn = var.role_arn

    # Inherit session tags from the configure-aws-credentials action
    transitive_tag_keys = [
      "GitHub",
      "Repository",
      "Workflow",
      "Action",
      "Actor",
      "Branch",
      "Commit"
    ]
  }

  default_tags {
    tags = local.tags
  }
}

variable "aws_profile" {
  type    = string
  default = null
}
variable "role_arn" {
  type    = string
  default = null
}
variable "shared_credentials_file" {
  type    = string
  default = "~/.aws/credentials"
}
variable "vault_address" {
  default = null
}
```
